R/commonutils.R

```diff
@@ -1,3 +1,27 @@
+#' Check if a R package is installed.
+#' If installed, load the library;
+#' otherise, stop and issue warning message
+#'
+#' @param pkgNmae Package Name, a string
+#' @param level Message levels: \cr
+#' 1: message \cr
+#' 2: warn \cr
+#' 3: stop \cr
+#' @return A boolean, if the package exists.
+#' @export
+checkPackage <- function(pkgName, level) {
```

Comment:
I thought about using `require`,  the main goal here is to give the user flexibility to issue message/warning/stop. Some of Dan's code only issues warning and default to certain solution. I know at least in some of my use case, I'd like it to just stop and quit R.